lib/datastore/entity.js

```diff
@@ -22,23 +22,71 @@
 'use strict';
 
 var arrify = require('arrify');
+var createErrorClass = require('create-error-class');
 var is = require('is');
 
-/** @const {object} Map for query operation -> operation protocol value. */
-var OP_TO_OPERATOR = {
-  '=':  'EQUAL',
-  '>':  'GREATER_THAN',
-  '>=': 'GREATER_THAN_OR_EQUAL',
-  '<':  'LESS_THAN',
-  '<=': 'LESS_THAN_OR_EQUAL',
-  HAS_ANCESTOR: 'HAS_ANCESTOR'
-};
-
-/** @const {object} Conversion map for query sign -> order protocol value. */
-var SIGN_TO_ORDER = {
-  '-': 'DESCENDING',
-  '+': 'ASCENDING'
-};
+var entity = module.exports;
+
+var InvalidKeyError = createErrorClass('InvalidKey', function(opts) {
+  var errorMessages = {
+    MISSING_KIND: 'A key should contain at least a kind.',
+    MISSING_ANCESTOR_ID: 'Ancestor keys require an id or name.'
+  };
+
+  this.message = errorMessages[opts.code];
+});
+
+/**
+ * Build a Datastore Double object.
+ *
+ * @constructor
+ * @param {number} value - The double value.
+ *
+ * @example
+ * var aDouble = new Double(7.3);
```

Comment:
This is kind of a general question in regards to this whole file - I'm noticing that in all the examples we're not referencing an entity, is this going to make the doc tests fail since none of these functions exist globally?